kernel/include/ksched.h

```diff
@@ -64,6 +64,7 @@ void z_ready_thread(struct k_thread *thread);
 void z_requeue_current(struct k_thread *curr);
 struct k_thread *z_swap_next_thread(void);
 void z_thread_abort(struct k_thread *thread);
+void z_sched_wake_thread(struct k_thread *thread, bool is_timeout);
```

Comment:
This looks like a good addition to the "official scheduler internal API" layer.  Can you move the declaration to the bottom of the file, near z_sched_wake() and write a proper documentation block for it?